### DISCUSSION TITLE
Laser si blocca per una frazione di secondo

### DISCUSSION THREAD

**Discussion by Daniel-Costanzi** [arkypita/LaserGRBL#1103] [Q&A]
Ciao! Da poco sto avendo un problema durante l'incisione. In pratica mi sono reso conto che alla fine di ogni linea che viene incisa il la testa laser si blocca per una frazione di secondo rendendo i bordi delle immagini più "bruciati" rispetto al resto. Fino a pochi giorni fa non mi sembrava che ci fosse questo problema e non capisco proprio a cosa potrebbe essere dovuto.
Inoltre, ho notato che succede solo quando le linee sono lunghe e molto frammentate (cioè nelle zone in cui le immagini sono grandi e cariche di dettagli).
Qualche idea di quale potrebbe essere la causa?
Grazie in anticipo e complimenti per il lavoro fatto.

DC

**Comment by BeatSlayer** [comment]
Most probably the speed it's too high for the detail amount.

Try a lower engraving speed.

**Comment by Daniel-Costanzi** [comment]
I noticedthat it has always happened, also last days, but now is more marked and the setting is the same.
I also tried to reduce engraving speed and laser power but results are almost unchanged.

**Comment by StuartB4** [comment]
Hai la modalità Laser abilitata nella configurazione grbl. In caso di dubbi, avviare LaserGRBL e fare clic su Connetti. Vai al menu Grbl e fai clic su Configurazione Grbl e guarda l'elenco per $ 32 e $ 30. Impostare $ 32 su 1 e $ 30 su 255. Quindi fare clic sul pulsante Scrivi in ​​basso e Chiudi. Usa anche M4 per Laser acceso nelle impostazioni dell'immagine di destinazione.

![image](https://user-images.githubusercontent.com/16980026/90981916-ac140800-e55b-11ea-8d9c-b274974c4c24.png)

![image](https://user-images.githubusercontent.com/16980026/90981958-fac1a200-e55b-11ea-94a7-91ca25b06dc9.png)

**Comment by arkypita** [comment]
questione di configurazione. vedi il suggerimento di @StuartB4

**Comment by Daniel-Costanzi** [comment]
What is the difference between M3 and M4?

Thank you for your help.

**Comment by StuartB4** [comment]
M3 Constant Laser Power Mode:
Constant laser power mode simply keeps the laser power as programmed, regardless if the machine is moving, accelerating, or stopped. This provides better control of the laser state. With a good G-code program, this can lead to more consistent cuts in more difficult materials.

For a clean cut and prevent scorching with M3 constant power mode, it's a good idea to add lead-in and lead-out motions around the line you want to cut to give some space for the machine to accelerate and decelerate.
NOTE: M3 can be used to keep the laser on for focusing.

M4 Dynamic Laser Power Mode:
Dynamic laser power mode will automatically adjust laser power based on the current speed relative to the programmed rate. It essentially ensures the amount of laser energy along a cut is consistent even though the machine may be stopped or actively accelerating. This is very useful for clean, precise engraving and cutting on simple materials across a large range of G-code generation methods by CAM programs. It will generally run faster and may be all you need to use.

 Grbl calculates laser power based on the assumption that laser power is linear with speed and the material. Often, this is not the case. Lasers can cut differently at varying power levels and some materials may not cut well at a particular speed and/power. In short, this means that dynamic power mode may not work for all situations. Always do a test piece prior to using this with a new material or machine.

When not in motion, M4 dynamic mode turns off the laser. It only turns on when the machine moves. This generally makes the laser safer to operate, because, unlike M3, it will never burn a hole through your table, if you stop and forget to turn M3 off in time.

**Comment by Daniel-Costanzi** [comment]
Thank you very much for the useful informations!! Setting M4 instead of M3 solved the problem.

Last question, what is the difference between setting $30=255 or 1000?

**Comment by StuartB4** [comment]
It's just another way of setting the maximum power for the PWM.
If you set it to 255 in Grbl and then adjust the max power in Target image settings when preparing the image for burning, setting it to 128 would be half the Max power.
If it was set to 1000 in Grbl and then adjusted to 500 in Target image settings that would also be half power.
